src/Adapter/Image/Uploader/ProductImageUploader.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use ErrorException;
+use Hook;
+use Image;
+use PrestaShop\PrestaShop\Adapter\Image\Exception\CannotUnlinkImageException;
+use PrestaShop\PrestaShop\Adapter\Product\ProductImagePathFactory;
+use PrestaShop\PrestaShop\Core\Configuration\UploadSizeConfigurationInterface;
+
+//@todo: do we really need an interface? depends if we use it in controller or in command handler
```

Comment:
In case of doubt: yes, let's have an interface.

Reply:
If we see it as all other Product services that we were recently introducing,
then Im more in favour to NOT use the interface and use the uploader inside handler, as its implemented now.
Product images command should even be testable because it doesn't use the `move_uploaded_file`. (It uses ImageManager:resize to put the "optimized' image where it belongs)